os/fs/driver/mtd/smart.c

```diff
@@ -1005,6 +1033,48 @@ static int smart_setsectorsize(FAR struct smart_struct_s *dev, uint16_t size)
 			dev->availSectPerBlk = dev->sectorsPerBlk;
 		}
 	}
+	
+#ifdef CONFIG_MTD_SMART_JOURNALING
+	/** Journal Sector is reserved at the last of smartfs partition, it doesn't use MTD Header.
+	  * Journal sectors divided to 2 area, this will be toggled if anyone is filled with data,
+	  * erase previous one. So total number of journal sector must be N*Eraseblock
+	  * We will use it as a contigous memory space...
+	  */
+	size_t log_size = sizeof(struct smart_journal_logging_data_s);
+	/** First get ideal size of journal sector. Formula is Below (x : data sectors y : journal sectors)
+	  * 1. dev->sectorsize : dev->njournalsectors = x : y  
+	  * 2. x + y = dev->geo.eraseblocks * dev->sectorsPerBlk;
+	  */
+	dev->njournalsectors = (dev->geo.neraseblocks * dev->sectorsPerBlk * log_size) / (dev->sectorsize + log_size);
```

Comment:
How does this equation work?
The numerator and denominator should mean something.

Numerator = (dev->geo.neraseblocks * dev->sectorsPerBlk * log_size) = **(TOTAL NUMBER OF SECTORS * log_size)**
= ? (This product represents what)

Denominator = (dev->sectorsize + log_size) = ? (This sum represents what)

Also, how is this equation derived from the above two formulas?

Reply:
See above comment. that typo error happened because I changed name of dev structure, and add new one.

Reply:
Now the origin of this equation becomes clear.